### PR TITLE
Update to v24 upstream, minor build edits, add Full RBF config option

### PR DESCRIPTION
**Pull request by kn0wmad**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -103,16 +104,16 @@ ENV BITCOIN_PREFIX=/opt/bitcoin-${BITCOIN_VERSION}
 RUN wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/SHA256SUMS
 RUN wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/SHA256SUMS.asc
 RUN wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/bitcoin-${BITCOIN_VERSION}.tar.gz
-RUN patch -u SHA256SUMS.asc -i SHA256SUMS.asc.patch
+# RUN patch -u SHA256SUMS.asc -i SHA256SUMS.asc.patch
```
Comment (chrisguida) [inline Dockerfile:107]: If we don't need the patch file, let's delete the patch file. Also just delete this line, don't comment.

File: Dockerfile [hunk 2]
```diff
@@ -103,16 +104,16 @@ ENV BITCOIN_PREFIX=/opt/bitcoin-${BITCOIN_VERSION}
...
 RUN gpg --verify SHA256SUMS.asc
 RUN grep " bitcoin-${BITCOIN_VERSION}.tar.gz\$" SHA256SUMS | sha256sum -c -
 RUN tar -xzf *.tar.gz
 
 WORKDIR /bitcoin-${BITCOIN_VERSION}
 
-RUN sed -i '/AC_PREREQ/a\AR_FLAGS=cr' src/univalue/configure.ac
+# RUN sed -i '/AC_PREREQ/a\AR_FLAGS=cr' src/univalue/configure.ac
```
Comment (chrisguida) [inline Dockerfile:114]: why was this added originally?

also delete if not needed

File: Dockerfile [hunk 3]
```diff
@@ -103,16 +104,16 @@ ENV BITCOIN_PREFIX=/opt/bitcoin-${BITCOIN_VERSION}
...
 RUN sed -i '/AX_PROG_CC_FOR_BUILD/a\AR_FLAGS=cr' src/secp256k1/configure.ac
-RUN sed -i s:sys/fcntl.h:fcntl.h: src/compat.h
+# RUN sed -i s:sys/fcntl.h:fcntl.h: src/compat.h
```
Comment (chrisguida) [inline Dockerfile:116]: same as above, why was this originally here?

File: manifest.yaml [hunk 4]
```diff
@@ -1,8 +1,11 @@
 id: bitcoind
 title: "Bitcoin Core"
-version: 23.0.1
+version: 24.0.0
 release-notes: |
     * Add build for x86
+    * Use new eOS 0.3.3 APIs
```
Comment (chrisguida) [inline manifest.yaml:6]: This line not needed, unless there's some js conversion i missed

File: migrations/gt_22_0_0_lt_23_0_0.sh [hunk 5]
```diff
@@ -5,15 +5,16 @@ set -ea
...
 elif [ $1 = "to" ]; then
     yq -i '.advanced.peers.addnode |= map_values(.hostname)' /root/.bitcoin/start9/config.yaml
     yq -i 'del(.advanced.bloomfilters)' /root/.bitcoin/start9/config.yaml
+    yq -i 'del(.advanced.mempool.mempoolfullrbf)' /root/.bitcoin/start9/config.yaml
     echo '{"configured": false }'
     exit 0
 else
     echo "FATAL: Invalid argument: {from, to}. Migration failed." >&2
     exit 1
 fi
```
Comment (chrisguida) [inline migrations/gt_22_0_0_lt_23_0_0.sh:20]: Make it a blank line, as is customary for all checked-in files

```suggestion
fi

```

File: scripts/services/getConfig.ts [hunk 6]
```diff
@@ -116,7 +116,7 @@ export const getConfig: T.ExpectedExports.getConfig = compat.getConfig({
               "Set the depth of the work queue to service RPC calls. Determines how long the backlog of RPC requests can get before it just rejects new ones.",
             "type": "number",
             "nullable": false,
-            "default": 64,
+            "default": 128,
```
Comment (chrisguida) [inline scripts/services/getConfig.ts:119]: Have we determined that 128 is better than 64?

File: scripts/services/getConfig.ts [hunk 7]
```diff
@@ -178,6 +178,13 @@ export const getConfig: T.ExpectedExports.getConfig = compat.getConfig({
         "name": "Mempool",
         "description": "Mempool Settings",
         "spec": {
+          "mempoolfullrbf": {
+            "name": "Enable Full RBF",
+            "description":
+              "Policy for your node to use for relaying and mining unconfirmed transactions.  For details, see https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-24.0.md",
```
Comment (chrisguida) [inline scripts/services/getConfig.ts:184]: Link to the actual section where it mentions full rbf?

```suggestion
              "Policy for your node to use for relaying and mining unconfirmed transactions.  For details, see https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-24.0.md#notice-of-new-option-for-transaction-replacement-policies",
```

**Review by chrisguida — CHANGES_REQUESTED** [review]
Just some questions and minor nits

File: Dockerfile (hunk 2 above)
Comment (kn0wmad) [reply Dockerfile:114]: I did not create this dockerfile, we'd have to ask @dr-bonez most likely

File: Dockerfile (hunk 3 above)
Comment (kn0wmad) [reply Dockerfile:116]: Same as above

File: Dockerfile (hunk 1 above)
Comment (kn0wmad) [reply Dockerfile:107]: I left all this stuff for the review, removing

File: scripts/services/getConfig.ts (hunk 6 above)
Comment (kn0wmad) [reply scripts/services/getConfig.ts:119]: This preference was given by @MattDHill and @BitcoinMechanic - I'm not sure how deep the testing went on it.  I'm not convinced it solves anything, but it also probably won't hurt anything

File: Dockerfile (hunk 1 above)
Comment (chrisguida) [reply Dockerfile]: The review will show all changes, no need to do this in the future :)

File: Dockerfile (hunk 2 above)
Comment (chrisguida) [reply Dockerfile]: In that case, what does it fix?

File: Dockerfile (hunk 3 above)
Comment (chrisguida) [reply Dockerfile]: In that case, what does it fix?

File: Dockerfile (hunk 2 above)
Comment (kn0wmad) [reply Dockerfile]: As I said, the build fails with these 2 lines included

Comment (chrisguida) [reply Dockerfile]: Yes, I'm looking for something more specific than "the build fails". What fails? What's the error message you're seeing?

File: Dockerfile (hunk 1 above)
Comment (kn0wmad) [reply Dockerfile]: This is such that I can immediately replace the line if necessary

File: Dockerfile (hunk 2 above)
Comment (chrisguida) [reply Dockerfile]: Yeah maybe we should check with @dr-bonez , the file not existing does not mean that it's ok to delete this change. Perhaps the file was simply moved upstream

**Review by chrisguida — APPROVED** [review]
Looks good

**Pull request merged**